src/Fuzz.elm

```diff
@@ -452,158 +396,62 @@ array fuzzer =
     map Array.fromList (list fuzzer)
 
 
-{-| Turn a tuple of fuzzers into a fuzzer of tuples.
+{-| Map over two fuzzers.
 -}
-tuple : ( Fuzzer a, Fuzzer b ) -> Fuzzer ( a, b )
-tuple ( Internal.Fuzzer baseFuzzerA, Internal.Fuzzer baseFuzzerB ) =
-    Internal.Fuzzer
-        (\noShrink ->
-            case ( baseFuzzerA noShrink, baseFuzzerB noShrink ) of
-                ( Gen genA, Gen genB ) ->
-                    Gen <| Random.map2 (,) genA genB
-
-                ( Shrink genTreeA, Shrink genTreeB ) ->
-                    Shrink <| Random.map2 tupleShrinkHelp genTreeA genTreeB
-
-                ( a, b ) ->
-                    [ invalidReason a, invalidReason b ]
-                        |> List.filterMap identity
-                        |> String.join " "
-                        |> InvalidFuzzer
-        )
-
+map2 : (a -> b -> c) -> Fuzzer a -> Fuzzer b -> Fuzzer c
+map2 transform fuzzA fuzzB =
+    (Result.map2 << Random.map2) (map2ShrinkHelp transform) fuzzA fuzzB
```

Comment:
I based `tuple` on this rather than the other way around because `map2` was easier to compose with.